spiketoolkit/preprocessing/transform.py

```diff
@@ -40,10 +42,28 @@ def get_traces(self, channel_ids=None, start_frame=None, end_frame=None):
         if channel_ids is None:
             channel_ids = self.get_channel_ids()
         traces = self._recording.get_traces(channel_ids=channel_ids, start_frame=start_frame, end_frame=end_frame)
-        return traces*self._scalar + self._offset
+        if isinstance(self._scalar, (int, float, np.integer, np.float)):
+            traces = traces*self._scalar
+        else:
+            if len(self._scalar) == len(channel_ids):
+                scalar = self._scalar
+            else:
+                channel_idxs = np.array([self._recording.get_channel_ids().index(ch) for ch in channel_ids])
+                scalar = np.array(self._scalar)[channel_idxs]
+            traces = np.array([t * s for (t, s) in zip(traces, scalar)])
```

Comment:
No. Do it it the vector numpy way please.

Reply:
yeah they ar both arrays. it should work